image-fetcher/fetch-image.sh

```diff
@@ -1,10 +1,16 @@
 #!/bin/sh
 set -e
 
-gcloud -q auth activate-service-account \
-  --key-file=/secrets/gcr-pull.json
+gcloud_auth() {
+    gcloud -q auth activate-service-account --key-file=/secrets/gcr-pull.json
+    gcloud -q auth configure-docker gcr.io
+}
 
-gcloud -q auth configure-docker
+if ! gcloud_auth > gcloud-auth.log 2>&1; then
```

Comment:
My bash knowledge isn't as good as yours. Do you mind explaining this to me?

Reply:
`2>&1` redirects everything that `gcloud_auth` would send to its stderr (2) to its stdout (1), so that line means "send everything `gcloud_auth` would output to that log file". Then if anything fails, we can capture any and all output from `gcloud` and log it as an error.

I am realizing though that this is slightly incorrect. I need to include `&&` between the two gcloud commands as right now if the first one fails and the second passes `gcloud_auth` will pass. Will push a fix.